container-gcp-csharp/Program.cs

```diff
@@ -19,12 +22,27 @@
     var location = gcpConfig.Require("region");
     var project = gcpConfig.Require("project");
 
+    // Create an Artifact Registry repository
+    var repository = new Gcp.ArtifactRegistry.Repository("my-repo", new()
+    {
+        Description = "Repository for container image",
+        Format = "DOCKER",
+        Location = location,
+        RepositoryId = "my-repo",
+    });
+
+    // Form the repository URL
+    var repoUrl = $"{location}-docker.pkg.dev/{project}/my-repo";
```

Comment:
It's probably worth pointing out that this failed the first time I ran it because `us-central1-docker.pkg.dev/my-teams-shared-project/my-repo` had already been created by someone else:

```
gcp:artifactregistry:Repository (my-repo):
    error: 1 error occurred:
        * Error creating Repository: googleapi: Error 409: the repository already exists
```

I'm not sure how folks use GCP in practice, but if projects are commonly shared among multiple developers, I suspect this'll fail pretty regularly -- probably even for us, when we start running templates in automated tests.

Since the assumption seems to be that the repo will be managed by the Pulumi program, should we consider naming the repo somehow randomly to avoid collisions? What do you think?

Reply:
You're right. Unfortunately, we can't use Pulumi's auto-naming functionality here. However, I've implemented a random string approach (in commits 9628dd5 and 75ba1c3) that should help. PTAL!